android/tangram/jni/platform_android.cpp

```diff
@@ -245,20 +256,25 @@ unsigned char* bytesFromFileSystem(const char* _path, unsigned int* _size) {
 
 }
 
-std::string stringFromFile(const char* _path, PathType _type) {
+std::string stringFromFile(const char* _path, PathType _type, const char* _resourceRoot) {
 
     unsigned int length = 0;
-    unsigned char* bytes = bytesFromFile(_path, _type, &length);
+    unsigned char* bytes = bytesFromFile(_path, _type, &length, _resourceRoot);
     std::string out(reinterpret_cast<char*>(bytes), length);
     free(bytes);
 
     return out;
 
 }
 
-unsigned char* bytesFromFile(const char* _path, PathType _type, unsigned int* _size) {
+unsigned char* bytesFromFile(const char* _path, PathType _type, unsigned int* _size, const char* _resourceRoot) {
```

Comment:
I agree on it.
To start with I removed I removed a Texture constructor which was kinda redundant as a Texture should only care about the blob of texture data and not the file/resource storing that data.